packages/apps/transfer/src/services/cross-chain-transfer-finish/get-transfer-data.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  ChainwebNetworkId,
+  ICommandResult,
+} from '@kadena/chainweb-node-client';
+import { PactCommand } from '@kadena/client';
+import { ChainId, IPactExec } from '@kadena/types';
+
+import {
+  convertIntToChainId,
+  generateApiHost,
+  validateRequestKey,
+} from '../utils/utils';
+
+import { Translate } from 'next-translate';
+interface ITransactionData {
+  sender: { chain: ChainId; account: string };
+  receiver: { chain: ChainId; account: string };
+  amount: number;
+  receiverGuard: {
+    pred: string;
+    keys: [string];
+  };
+  step: number;
+  pactId: string;
+  rollback: boolean;
+}
+export interface ITransferDataResult {
+  tx?: ITransactionData | undefined;
+  error?: string;
+}
+
+export interface ISpvProofResult {
+  proof?: string;
+  error?: string;
+}
+
+export async function getTransferData({
+  requestKey,
+  server,
+  networkId,
+  t,
+}: {
+  requestKey: string;
+  server: string;
+  networkId: ChainwebNetworkId;
+  t: Translate;
+}): Promise<ITransferDataResult> {
+  const validatedRequestKey = validateRequestKey(requestKey);
+
+  if (validatedRequestKey === undefined) {
+    return { error: t('Invalid length of request key') };
+  }
+
+  const pactCommand = new PactCommand();
+
+  pactCommand.requestKey = requestKey;
+
+  try {
+    const chainInfoPromises = Array.from(new Array(20)).map((item, chainId) => {
```

Comment:
I'd rather have a constant with an array with all the separate chains spelled out. There's no reason chains will be always just a number. Maybe we can fix this in a separate PR across this project